### PR TITLE
Beta/hotfix tooling

### PR DESCRIPTION
**Pull request by loremattei**
This PR adds two lanes to automate the versioning for new beta and hotfix releases.
There's a related PR in release-toolkit: wordpress-mobile/release-toolkit#57.
This PR builds on top of https://github.com/woocommerce/woocommerce-android/pull/1225 and should be reviewed/merged after it.

# To test:
### 1. Beta
The normal flow for this command is to call it without passing a version. It should read it from develop and update the related branch.
Since the code will commit and push on the release branch, we'll override the release branch name to make it push to a different branch.

1.1 Checkout a new `release/2.3` branch and push it with the `-u` option. (_Note 1_: You can actually call it with any number you want, the code only checks that it is a release branch. _Note 2_: You want to checkout a new branch, because the code will commit and push to the current one).
1.2 Run `bundle exec fastlane new_beta_release base_version:2.3` (You should use the same version you checked out above).
1.3 Verify that the code reads the correct versions from `gradle.properties`. It should read the one from develop and the one from `release/2.3`. They are probably the same as the branches are in sync now.
1.4 Verify that the beta name is correctly updated, the changes committed and pushed and the new version tagged. Note: The code should increment the current beta version number, currently `2.2-rc-1` to `2.2-rc-2` and don't check the base version (2.2) against the release branch name (2.3). The check is intentionally disabled because this is an override feature and we may want a bit more freedom.
1.5 Delete the new beta tag from your local repository and from GitHub.
1.6 Delete the `release/2.3` branch from your local repository and from GitHub.

### 2. Hotfix
2.1 Add a `2.2` tag to your local repository (no need to push it).
2.2 Run `bundle exec fastlane new_hotfix_release version_name:2.2.1 version_code:80`. You can choose any 2.2.x version and any version code.
2.3 Verify that the code calculates the right base version.
2.4 Verify that the code checks out a new `release/2.2.1` branch from your 2.2 tag.
2.5 Verify that the branch is pushed.
2.6 Verify that the version is correctly updated in `gradle.properties`.
2.7 Verify that the version is tagged.
2.8 Delete the new 2.2.1 tag from you local repository and from GitHub.
2.9 Delete the new `release/2.2.1` branch from your local repository and from GitHub.
2.10 Delete the 2.2 tag from your local repository.

Note: When approved, I'll update the reference to the release-toolbox before merging this to develop.

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by jtreanor** [comment]
# To test:
### 1. Beta

- [ ] Checkout a new `release/2.4` branch and push it with the `-u` option. (_Note 1_: You can actually call it with any number you want, the code only checks that it is a release branch. _Note 2_: You want to checkout a new branch, because the code will commit and push to the current one).
- [ ] Run `bundle exec fastlane new_beta_release base_version:2.4` (You should use the same version you checked out above).
- [ ] Verify that the code reads the correct versions from `gradle.properties`. It should read the one from develop and the one from `release/2.4`. They are probably the same as the branches are in sync now.
- [ ] Verify that the beta name is correctly updated, the changes committed and pushed and the new version tagged. Note: The code should increment the current beta version number, currently `2.3-rc-1` to `2.3-rc-2` and don't check the base version (2.3) against the release branch name (2.4). The check is intentionally disabled because this is an override feature and we may want a bit more freedom.
- [ ] Delete the new beta tag from your local repository and from GitHub.
- [ ] Delete the `release/2.4` branch from your local repository and from GitHub.

### 2. Hotfix
- [ ] Add a `2.3` tag to your local repository (no need to push it).
- [ ] Run `bundle exec fastlane new_hotfix_release version_name:2.3.1 version_code:80`. You can choose any 2.3.x version and any version code.
- [ ] Verify that the code calculates the right base version.
- [ ] Verify that the code checks out a new `release/2.3.1` branch from your 2.3 tag.
- [ ] Verify that the branch is pushed.
- [ ] Verify that the version is correctly updated in `gradle.properties`.
- [ ] Verify that the version is tagged.
- [ ] Delete the new 2.3.1 tag from you local repository and from GitHub.
- [ ] Delete the new `release/2.3.1` branch from your local repository and from GitHub.
- [ ] Delete the 2.3 tag from your local repository.

File: fastlane/Fastfile [hunk 1]
```diff
@@ -125,9 +239,65 @@ lane :get_pullrequests_list do | options |
...
+    aab_file="WooCommerce.aab"
+    output_dir="WooCommerce/build/outputs/bundle/"
+    build_dir="artifacts/"
+    logfile_path="#{build_dir}build.log"
+
+    # Build
+    Dir.chdir("..") do
+      UI.message("Cleaning branch...")
+      sh("echo \"Cleaning branch\" >> #{logfile_path}")
+      sh("./gradlew clean >> #{logfile_path} 2>&1")
```
Comment (jtreanor) [inline fastlane/Fastfile]: Can we use the `gradle` action here as you did in the other PR?

Comment (loremattei) [reply fastlane/Fastfile]: Yep! This refers to the commit on the other PR, so it should be updated when merging.

Comment (loremattei) [reply fastlane/Fastfile]: Anyway, I merged `develop` back into this branch to confirm this :-) .

**Comment by loremattei** [comment]
Thanks for reviewing this!

**Pull request merged**